### PR TITLE
Remove usage of global variable "inlined_types"

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @pietroalbini

### CODE REVIEW

**Comment by pietroalbini** [comment]
Thanks for the PR! I don't feel I'm a good reviewer for it though (never touched rustdoc myself).

**Comment by GuillaumeGomez** [comment]
So let's go for @kinnison then!

r? @kinnison 

cc @ollie27 as well. :)

**Comment by GuillaumeGomez** [comment]
Fixed the tidy issue.

File: src/librustdoc/html/render.rs [hunk 1]
```diff
@@ -992,13 +992,11 @@ themePicker.onblur = handleThemeButtonsBlur;
...
-                window.register_implementors(implementors);
-            } else {
-                window.pending_implementors = implementors;
-            }
-        ",
+            "if (window.register_implementors) {\
+                 window.register_implementors(implementors);\
+             } else {\
+                 window.pending_implementors = implementors;\
+             }",
```
Comment (kinnison) [inline src/librustdoc/html/render.rs]: This change makes it harder to read the string.  Why did it happen?

File: src/librustdoc/html/render.rs [hunk 2]
```diff
@@ -2700,17 +2704,12 @@ fn item_trait(w: &mut Buffer, cx: &Context, it: &clean::Item, t: &clean::Trait)
...
-    );
 
     write!(
         w,
-        r#"<script type="text/javascript" async
-                         src="{root_path}/implementors/{path}/{ty}.{name}.js">
-                 </script>"#,
+        "<script type=\"text/javascript\" \
+                 src=\"{root_path}/implementors/{path}/{ty}.{name}.js\" async>\
+         </script>",
```
Comment (kinnison) [inline src/librustdoc/html/render.rs]: Ditto this change makes it harder to read IMO.

**Review by kinnison — COMMENTED** [review]
I see the principal of what is going on here, and other than the two less nice to read strings I'm OK with the implementation.  I'm going to assume we have some tests covering the presence of synthetic impls, but is it worth writing a test which ensures the `aliases=...` attribute is appearing properly?

File: src/librustdoc/html/render.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/render.rs]: What was harder to read? The change itself or the generated code? If it's the latter, it's on purpose.

File: src/librustdoc/html/render.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/render.rs]: To be more consistent with the rest of the code. Everywhere else, we inline the output HTML/JS to reduce its size.

**Comment by GuillaumeGomez** [comment]
About the `aliases` check, it has do be done inside a browser to check that the UI doesn't change. However, I'll add a test to check that the attribute is correctly generated.

File: src/librustdoc/html/render.rs (hunk 2 above)
Comment (kinnison) [reply src/librustdoc/html/render.rs]: I mean the change from an `r#".."#` to a `"...."` string -- which ends up with escaped quotes.

File: src/librustdoc/html/render.rs (hunk 1 above)
Comment (kinnison) [reply src/librustdoc/html/render.rs]: As with the other, it's that the change from a raw string to a non raw string means those backslashes are a pain.

File: src/librustdoc/html/render.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/render.rs]: It's same the everywhere else and it has the advantage to generate much smaller strings (which is the primary goal). At least I don't see escaped quotes as an issue.

Comment (kinnison) [reply src/librustdoc/html/render.rs]: OK fair enough.

**Comment by GuillaumeGomez** [comment]
Added the test.

**Review by kinnison — COMMENTED** [review]
As far as I can tell, the UI is not broken by this -- I've reviewed a number of crates' docs which I'm at least passingly familiar with and things look sane.

@bors r+

**Comment by GuillaumeGomez** [comment]
@kinnison You need to comment apparently, bors doesn't detect on review comments I think.

@bors: r=kinnison

**Pull request merged**